### PR TITLE
#38259: spanish translation poc

### PR DESCRIPTION
**Pull request by acrollet**
## Original issue(s)
department-of-veterans-affairs/va.gov-team#38259

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/check-in/components/pages/ConfirmablePage/index.jsx [hunk 1]
```diff
@@ -39,23 +44,36 @@ const ConfirmablePage = ({
                 {field.key in data && data[field.key] ? (
                   <DemographicItem demographic={data[field.key]} />
                 ) : (
-                  'Not available'
+                  t('not-available')
+                )}
+                {isEditEnabled && field.editAction && (
```
Comment (brianseek) [inline src/applications/check-in/components/pages/ConfirmablePage/index.jsx:49]: Weird linting errors below. whitespace issue? I had a PR with this issue a while back too.

File: src/applications/check-in/components/pages/Edit/Address.jsx [hunk 2]
```diff
@@ -90,7 +92,7 @@ export default function Address(props) {
         if (!newValue) {
           setAddressError(prevState => ({
             ...prevState,
-            [fieldName]: `${event.target.label} is required`,
+            [fieldName]: `${event.target.label} ${t('is-required')}`,
```
Comment (brianseek) [inline src/applications/check-in/components/pages/Edit/Address.jsx:95]: Looks like t needs to get added to the dependency array when used in a hook.

File: src/applications/check-in/components/pages/Edit/Relationship.jsx [hunk 3]
```diff
@@ -59,33 +56,33 @@ export default function Relationship(props) {
...
-      BROTHER: 'Brother',
-      SISTER: 'Sister',
-      MOTHER: 'Mother',
-      FATHER: 'Father',
+      SPOUSE: t('spouse'),
+      CHILD: t('child'),
+      BROTHER: t('brother'),
+      SISTER: t('sister'),
+      MOTHER: t('mother'),
+      FATHER: t('father'),
```
Comment (brianseek) [inline src/applications/check-in/components/pages/Edit/Relationship.jsx:64]: I just merged an update to this that changes these values. Probably need to do a rebase and fix them up again.

File: src/applications/check-in/components/pages/Edit/Relationship.jsx [hunk 4]
```diff
@@ -111,6 +108,7 @@ export default function Relationship(props) {
       />
       <Footer />
       <BackToHome />
+      <LanguagePicker />
```
Comment (brianseek) [inline src/applications/check-in/components/pages/Edit/Relationship.jsx:111]: Should we add the language picker once to the footer component?

File: src/applications/check-in/components/pages/nextOfKin/NextOfKinDisplay.jsx [hunk 5]
```diff
@@ -10,7 +12,7 @@ import { URLS } from '../../../utils/navigation';
 import { EDITING_PAGE_NAMES } from '../../../utils/appConstants';
 
 export default function NextOfKinDisplay({
-  header = 'Is this your current next of kin information?',
+  header = i18next.t('is-this-your-current-next-of-kin-information'),
```
Comment (brianseek) [inline src/applications/check-in/components/pages/nextOfKin/NextOfKinDisplay.jsx:15]: why are some translated like this with `i18next` and others using `react-i18next`?

File: src/applications/check-in/components/pages/validate/ValidateDisplay.jsx [hunk 6]
```diff
@@ -75,10 +77,14 @@ export default function ValidateDisplay({
         className="usa-button usa-button-big"
         data-testid="check-in-button"
         disabled={isLoading}
-        aria-label="Check in now for your appointment"
+        aria-label={i18next.t('check-in-now-for-your-appointment')}
       >
         {' '}
-        {isLoading ? <span role="status">Loading...</span> : <>Continue</>}
+        {isLoading ? (
+          <span role="status">Loading...</span>
```
Comment (brianseek) [inline src/applications/check-in/components/pages/validate/ValidateDisplay.jsx:84]: Missed the `t` on Loading...

File: src/applications/check-in/day-of/pages/CheckIn/DisplayMultipleAppointments.jsx [hunk 7]
```diff
@@ -62,8 +63,9 @@ const DisplayMultipleAppointments = props => {
           })}
         </ol>
         <p data-testid="update-text">
-          <strong>Latest update:</strong>{' '}
-          {format(new Date(), "MMMM d, yyyy 'at' h:mm aaaa")}
+          <Trans i18nKey="latest-update" values={{ date: new Date() }}>
```
Comment (brianseek) [inline src/applications/check-in/day-of/pages/CheckIn/DisplayMultipleAppointments.jsx:66]: Why does this one require the component wrapper?

**Review by brianseek — COMMENTED** [review]
@acrollet This looks really great. There are couple of pretty minor things I noticed, some is just linting stuff we should fix, components that need to be swapped out for webcomponents, and a couple of questions about translation implementation.

File: src/applications/check-in/components/PreCheckinConfirmation.jsx [hunk 8]
```diff
@@ -35,25 +38,29 @@ const PreCheckinConfirmation = props => {
...
         </p>
         <h3 data-testid="appointment-questions">
-          What if I have questions about my appointment?
+          {t('what-if-i-have-questions-about-my-appointment')}
         </h3>
-        <p>Call your VA health care team:</p>
+        <p>{t('call-your-va-health-care-team')}:</p>
         {appointments.map((appointment, index) => {
           return (
             <p key={index}>
```
[the remark this answers is not in the document]
Comment (acrollet) [reply src/applications/check-in/components/PreCheckinConfirmation.jsx:65]: Good call, done!

**Comment by acrollet** [comment]
> why are some translated like this with `i18next` and others using `react-i18next`? 

all fixed!

File: src/applications/check-in/components/pages/validate/ValidateDisplay.jsx (hunk 6 above)
Comment (acrollet) [reply src/applications/check-in/components/pages/validate/ValidateDisplay.jsx:88]: thanks!

File: src/applications/check-in/day-of/pages/CheckIn/DisplayMultipleAppointments.jsx (hunk 7 above)
Comment (acrollet) [reply src/applications/check-in/day-of/pages/CheckIn/DisplayMultipleAppointments.jsx]: It's needed when there are html tags in the translated string, c.f. https://react.i18next.com/latest/trans-component#usage-with-simple-html-elements-like-less-than-br-greater-than-and-others-v10.4.0

File: src/applications/check-in/components/pages/Edit/Address.jsx (hunk 2 above)
Comment (mdewey) [inline src/applications/check-in/components/pages/Edit/Address.jsx:95]: How is this handling the `{event.target.label}` translation?

**Review by mdewey — APPROVED** [review]
LGTM

File: src/applications/check-in/components/pages/Edit/Address.jsx (hunk 2 above)
Comment (brianseek) [reply src/applications/check-in/components/pages/Edit/Address.jsx:95]: that is a good catch, we probably need to hit our appConstants with translation.

Comment (acrollet) [reply src/applications/check-in/components/pages/Edit/Address.jsx:95]: yeah good call, looking at this 🙂

Comment (brianseek) [reply src/applications/check-in/components/pages/Edit/Address.jsx:95]: good point, yeah I think that could wait then

Comment (acrollet) [reply src/applications/check-in/components/pages/Edit/Address.jsx:95]: I made a note here: https://app.zenhub.com/workspaces/check-in-experience-61fc23a2cb8a14001132e102/issues/department-of-veterans-affairs/va.gov-team/38260

**Pull request merged**